### PR TITLE
Full implementation of spark-hdfs: Read/Write/Manage/Query-File 

### PR DESCRIPTION
**Pull request by rabbitonweb**
Implement all Read/Write/Manage/Query-File algebras

### CODE REVIEW

File: sparkcore/src/main/scala/quasar/physical/sparkcore/fs/hdfs/package.scala [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+      SparkHdfsFSDef(mapSNT[Eff, S](interpreter), lift(Task.delay(sc.stop())).into[S])
+    }).into[S]
+  }
+
+  private def fsInterpret(sparkFsConf: SparkFSConf): FileSystem ~> Free[Eff, ?] = {
+
+    def hdfsPathStr: AFile => Task[String] = (afile: AFile) => Task.delay {
+      sparkFsConf.hdfsUriStr + posixCodec.unsafePrintPath(afile)
+    }
+    // TODO consider closing over task
```
Comment (drostron) [inline sparkcore/src/main/scala/quasar/physical/sparkcore/fs/hdfs/package.scala]: Or `Read[HdfsFileSystem]`

File: sparkcore/src/main/scala/quasar/physical/sparkcore/fs/hdfs/managefile.scala [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+      delete.map(_.right[PhysicalError]).handle {
+        case NonFatal(e : Exception) => UnhandledFSError(e).left[FileSystemError \/ Unit]
+      }
+
+    Failure.Ops[PhysicalError, S].unattempt(lift(deleteHandled).into[S])
+  }
+  
+  private def tempFile[S[_]](near: APath)(implicit
+    s0: Task :<: S
+  ): Free[S, FileSystemError \/ AFile] = injectFT[Task, S].apply{
```
Comment (drostron) [inline sparkcore/src/main/scala/quasar/physical/sparkcore/fs/hdfs/managefile.scala]: `lift(...).into` is used everywhere else.

File: sparkcore/src/main/scala/quasar/physical/sparkcore/fs/hdfs/queryfile.scala [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+  def fileExists(f: AFile): Task[Boolean] =
+    toPath(f).map(path => {
+      val hdfs = fileSystem()
+      val exists = hdfs.exists(path)
+      hdfs.close()
+      exists
+    })
+
+  def listContents(d: ADir): EitherT[Task, FileSystemError, Set[PathSegment]] =
```
Comment (drostron) [inline sparkcore/src/main/scala/quasar/physical/sparkcore/fs/hdfs/queryfile.scala]: Common to use `FileSystemErrT`.

**Comment by rabbitonweb** [comment]
RDY

File: sparkcore/src/main/scala/quasar/physical/sparkcore/fs/hdfs/package.scala (hunk 1 above)
Comment (drostron) [inline sparkcore/src/main/scala/quasar/physical/sparkcore/fs/hdfs/package.scala]: I'd like to see this addressed as part of this PR. `Task` is acceptable but producing a `Read[FileSystem]` when `Task` is in the effect stack would follow usages elsewhere. I'm available to pair on that if helpful.

File: sparkcore/src/main/scala/quasar/physical/sparkcore/fs/hdfs/package.scala [hunk 4]
```diff
@@ -113,7 +113,7 @@ package object hdfs {
       sparkFsConf.hdfsUriStr + posixCodec.unsafePrintPath(afile)
     }
     // TODO consider closing over task or read 
-    def fileSystem: () => HdfsFileSystem = () => {
+    def fileSystem: () => Task[HdfsFileSystem] = () => Task.delay {
```
Comment (drostron) [inline sparkcore/src/main/scala/quasar/physical/sparkcore/fs/hdfs/package.scala]: Doesn't need to be a function anymore. Can just be:
```scala
val fileSystem: Task[HdfsFileSystem] = Task.delay { ...
```

**Comment by drostron** [comment]
Per conversation `fileSystem` will no longer be a function in an upcoming PR, so LGTM.

**Pull request merged**